### PR TITLE
Fix empty string bug + Improve code style in Dictionary 

### LINKED ISSUE #323: Empty string bug in ContextBar
To replicate:
Enter `nus     ` (i.e. some spaces after the `nus` keyword)
Bug: The event name shown on context bar is `assistant` (i.e. the smallest full phrase tracked by Awareness)

Adding extra spaces in between words creates more instances of `assistant` 

To fix:
Filter out the extra spaces between tokens, before performing the "slang - fullphrase lookup"

### PR DESCRIPTION
**Pull request by anubh-v**
Fixes #323 
Fixes the bug where adding spaces in `nus <space>`  finds `assistant` as the event name

### CODE REVIEW

File: src/main/java/seedu/address/model/awareness/Dictionary.java [hunk 1]
```diff
@@ -153,31 +159,36 @@ private String getFullPhraseFromPartialPhrase(String partialPhrase) {
...
-    /* Precondition: Input string has no trailing whitespaces */
-    private boolean isOneWord(String trimmedString) {
-        return trimmedString.split(SPACE).length == 1;
-    }
-
     private String[] tokenize(String expression) {
         return expression.split(SPACE);
     }
 
+    private String clean(String s) {
```
Comment (marvintxd) [inline src/main/java/seedu/address/model/awareness/Dictionary.java]: Can I ask why `trim()` should be wrapped? Seems like the previous way of directly calling was ok

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 2]
```diff
@@ -81,4 +81,23 @@ public static boolean isOnlyWhiteSpace(String s) {
         requireNonNull(s);
         return isEmptyString(s.trim());
     }
+
+    /**
+     * Returns true iff {@code s} has no trailing whitespaces.
+     */
+    public static boolean hasNoTrailingWhiteSpace(String s) {
```
[the remark this answers is not in the document]
Comment (anubh-v) [reply src/main/java/seedu/address/commons/util/StringUtil.java]: This is my mistake - I always assume trailing means both front and back.

I will rename to `isPaddedByWhiteSpace` ?

File: src/main/java/seedu/address/model/awareness/Dictionary.java (hunk 1 above)
Comment (anubh-v) [reply src/main/java/seedu/address/model/awareness/Dictionary.java]: This was my attempt to achieve SLAP. 
As shown below:
image

File: src/main/java/seedu/address/commons/util/StringUtil.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (anubh-v) [reply src/main/java/seedu/address/commons/util/StringUtil.java]: Done

**Pull request merged**